src/Pages/SignUp/SignUp.js

```diff
@@ -0,0 +1,381 @@
+import React from "react";
+import styled from "styled-components";
+import { Link } from "react-router-dom";
+import { useState, useEffect } from "react";
+import { signupAPI } from "../../config";
+
+const SignUp = () => {
+  const [userInfo, setUserInfo] = useState({
+    firstName: "",
+    lastName: "",
+    email: "",
+    userName: "",
+    password: "",
+  });
+
+  const [firstNameValid, setFirstNameValid] = useState(true);
+  const [emailValid, setEmailValid] = useState(true);
+  const [userNameValid, setUserNameValid] = useState(true);
+  const [passwordValid, setPasswordValid] = useState(true);
+
+  const handleInput = (name, e) => {
+    setUserInfo({ ...userInfo, [name]: e.target.value });
+  };
+
+  console.log("password: ", userInfo.password);
+
+  const isFirstNameValid = () => {
+    userInfo.firstName.length === 0
+      ? setFirstNameValid(false)
+      : setFirstNameValid(true);
+  };
+
+  const isEmailValid = () => {
+    !userInfo.email.includes("@") ? setEmailValid(false) : setEmailValid(true);
+  };
+
+  const isUserNameValid = () => {
+    userInfo.userName.length === 0
+      ? setUserNameValid(false)
+      : setUserNameValid(true);
+  };
+
+  const isPasswordValid = () => {
+    userInfo.password.length < 6
+      ? setPasswordValid(false)
+      : setPasswordValid(true);
+  };
+
+  const handleSignUp = () => {
+    if (firstNameValid && emailValid && userNameValid && passwordValid) {
+      fetch(signupAPI, {
+        method: "POST",
+        body: JSON.stringify({
+          firstName: userInfo.firstName,
+          lastName: userInfo.lastName,
+          email: userInfo.email,
+          userName: userInfo.userName,
```

Comment:
```suggestion
  const handleSignUp = () => {
    const { firstName, lastName, email, userName } = userInfo
    if (firstNameValid && emailValid && userNameValid && passwordValid) {
      fetch(signupAPI, {
        method: "POST",
        body: JSON.stringify({
          firstName, lastName,email, userName,
```
구조분해 할당 + 객체 단축 속성명 사용해주세요

Reply:
적용했습니다